src/index.js

```diff
@@ -0,0 +1,128 @@
+  lists.splice(index, 1);
+  localStorage.setItem('list', JSON.stringify(lists));
+  location.reload();
+};
+
+const editEvent = (id) => {
+  const index = id.slice(5).toString();
+  const textField = document.getElementById(`${index}-description`);
+  textField.innerHTML = '';
+  const editField = document.createElement('input');
+  editField.type = 'text';
+  editField.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      lists[index - 1].description = editField.value;
+      localStorage.setItem('list', JSON.stringify(lists));
+      textField.innerHTML = `${editField.value}`;
+    }
+  });
+  textField.appendChild(editField);
+};
+
+const updateTask = (item, index) => {
+  lists[index - 1].completed = item;
+  localStorage.setItem('list', JSON.stringify(lists));
+  const text = document.getElementById(`${index}-description`);
+  if (item) {
+    text.classList.add('overlined');
+  } else {
+    text.classList.remove('overlined');
+  }
+};
+
+function loadPredef(arr) {
+  for (let i = 0; i < arr.length; i += 1) {
+    setList(arr[i].description, arr[i].completed, counterIncreament());
+  }
+  localStorage.setItem('list', JSON.stringify(lists));
+  const completeCheckBox = document.querySelectorAll("input[type='checkbox']");
+  completeCheckBox.forEach((box) => {
+    box.addEventListener('click', () => {
+      updateTask(box.checked, box.value);
+    });
+  });
+  const completeRemovers = document.querySelectorAll('a.dropdown-remover');
+  completeRemovers.forEach((link) => {
+    link.addEventListener('click', () => {
+      deleteAll(link.id);
+    });
+  });
+  const completeEditors = document.querySelectorAll('a.dropdown-editor');
+  completeEditors.forEach((link) => {
+    link.addEventListener('click', () => {
+      editEvent(link.id);
+    });
+  });
+}
+
+loadPredef(lists);
+
+inputField.addEventListener('keypress', (e) => {
+  if (e.key === 'enter') {
+    const newIndex = counterIncreament();
+    const description = document.getElementById('new-task').value;
+
+    if (description === '' || description === ' ' || description == null) { return; }
+    setList(description, false, newIndex);
+    updateTask();
+
+    document.getElementById('new-item').value = '';
+    const newTask = new List(description, false, newIndex);
+
+    lists.push(newTask);
+    localStorage.setItem('list', JSON.stringify(lists));
+    location.reload();
+  }
+});
+
+addBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const newIndex = counterIncreament();
+  const description = document.getElementById('new-task').value;
+
+  if (description === '' || description === ' ' || description == null) { return; }
+  setList(description, false, newIndex);
+
+  document.getElementById('new-task').value = '';
+  const newTask = new List(description, false, newIndex);
+
+  lists.push(newTask);
+  localStorage.setItem('list', JSON.stringify(lists));
+  location.reload();
+});
```

Comment:
- [ ] Nice job you have done so far. However, I tried adding a task and I clicked on the `add button` but the task is not being added. I checked the console and found out that an error occurred. See the error below and let's try to fix this 💪 
<img width="925" alt="Loom_FLBVNkq3Zm" src="https://user-images.githubusercontent.com/81468675/149621667-e9eaafe1-ed04-49c6-90d5-bb3aa1721855.png">